test/box-luatest/downgrade_test.lua

```diff
@@ -973,3 +973,93 @@ g.test_downgrade_global_names = function(cg)
         t.assert_equals(#box.space._cluster:format(), 2)
     end)
 end
+
+-----------------------------
+-- Check downgrade from 3.1.0
+-----------------------------
+
+g.test_downgrade_func_trigger = function(cg)
+    cg.server:exec(function()
+        local helper = require('test.box-luatest.downgrade_helper')
+        local _func = box.space._func
+
+        local function check_space_func(has_trigger)
+            if has_trigger then
+                t.assert_equals(#_func:format(), 20)
+                t.assert_equals(_func:format()[20].name, 'trigger')
+            else
+                t.assert_equals(#_func:format(), 19)
+            end
+        end
+
+        local function check_func(func_names, event_name, has_trigger)
+            -- Allow to pass array of names and a single name
+            if type(func_names) ~= 'table' then
+                func_names = {func_names}
+            end
+            for _, func_name in pairs(func_names) do
+                local func_tuple = _func:get(box.func[func_name].id)
+                local func_tuple_len = 19
+                local trigger_field = nil
+                if has_trigger then
+                    func_tuple_len = 20
+                    trigger_field = {event_name}
+                end
+                t.assert_equals(func_tuple.trigger, trigger_field)
+                t.assert_equals(#func_tuple, func_tuple_len)
+            end
+        end
```

Comment:
Nit. I think test are more readable if they have explicit expectations.

We can have `before_downgrade` check and `after_downgrade` check as in other cases. Both have single flag argument whether we have trigger functions or not. No loops, branch only on argument.

Reply:
Done.